sunpy/data/_sample.py

```diff
@@ -51,52 +65,58 @@
     "NORH_LIGHTCURVE": ("tca110810", "")
 }
 
-sample_files = {}
-for key in _files:
-    sample_files[key] = os.path.abspath(os.path.join(sampledata_dir, _files[key][0]))
+
+free_required=145227776
+if disk_usage('/').free > free_required :
+
+	sample_files = {}
+	for key in _files:
+    		sample_files[key] = os.path.abspath(os.path.join(sampledata_dir, _files[key][0]))
+
+
+	def download_sample_data(progress=True, overwrite=True):
+		"""
+		Download the sample data.
+		Parameters
+		----------
+		progress: bool
+			Show a progress bar during download
+		overwrite: bool
+			If exist overwrites the downloaded sample data.
+		Returns
+		-------
+		None
+		"""
+		print("hello")
+		number_of_files_fetched = 0
+		print("Downloading sample files to {}".format(sampledata_dir))
+		for file_name in six.itervalues(_files):
+			if not overwrite:
+				if os.path.isfile(os.path.join(sampledata_dir,file_name[0])):
+					number_of_files_fetched += 1
+					continue
+			for base_url in _base_urls:
+				full_file_name = file_name[0] + file_name[1]
+				if url_exists(os.path.join(base_url, full_file_name)):
+					f = download_file(os.path.join(base_url, full_file_name))
+					real_name, ext = os.path.splitext(full_file_name)
+					if file_name[1] == '.zip':
+						print("Unpacking: {}".format(real_name))
+						with ZipFile(f, 'r') as zip_file:
+							zip_file.extract(real_name, sampledata_dir)
+						os.remove(f)
+					else:
+						# move files to the data directory
+						move(f, os.path.join(sampledata_dir, file_name[0]))
+					# increment the number of files obtained to check later
+					number_of_files_fetched += 1
+					break
+		if number_of_files_fetched < len(list(_files.keys())):
+			raise URLError("Could not download all samples files. Problem with accessing sample data servers.")
+else: 
+	print("insufficient diskspace")
```

Comment:
@hemantkhatri27  I think it would be better a `try-except` block than an `if` - this way the error can be caught by other routines.